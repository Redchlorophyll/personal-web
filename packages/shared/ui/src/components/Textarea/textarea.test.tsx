import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { useState } from "react";
import { Textarea } from "./index";
import { act } from "react-dom/test-utils";

function InputWrapper({
  disabled = false,
  limit,
}: {
  disabled?: boolean;
  limit?: number;
}) {
  const [value, setValue] = useState<string>("");

  const onChange = (event: React.FormEvent<HTMLTextAreaElement>): void => {
    setValue((event.target as HTMLTextAreaElement).value);
  };

  return (
    <div>
      <Textarea
        onChange={onChange}
        placeholder="This is Textarea"
        isDisabled={disabled}
        value={value}
        limit={limit}
        label="textarea"
      />
    </div>
  );
}

describe("shared - ui - Input", () => {
  test("It should render properly", () => {
    render(<InputWrapper />);
    const target = screen.getByRole("textbox");

    expect(target).toBeInTheDocument();
  });

  test("It should not allow text input when disabled", async () => {
    const user = userEvent.setup();
    render(<InputWrapper disabled={true} />);
    const target = screen.getByRole("textbox");

    await act(async () => await user.type(target, "test"));
    expect(screen.queryByDisplayValue("test")).not.toBeInTheDocument();
  });

  test("It should show right input when inputing value", async () => {
    const user = userEvent.setup();
    render(<InputWrapper />);
    const target = screen.getByRole("textbox");

    expect(screen.queryByDisplayValue("")).toBeInTheDocument();
    await act(async () => await user.type(target, "test"));
    expect(screen.queryByDisplayValue("test")).toBeInTheDocument();
  });

  test("It should show right input when inputing value", async () => {
    const user = userEvent.setup();
    render(<InputWrapper limit={5} />);
    const target = screen.getByRole("textbox");

    expect(screen.queryByDisplayValue("")).toBeInTheDocument();
    await act(async () => await user.type(target, "dhonni"));
    expect(screen.queryByDisplayValue("dhonn")).toBeInTheDocument();
  });
});
