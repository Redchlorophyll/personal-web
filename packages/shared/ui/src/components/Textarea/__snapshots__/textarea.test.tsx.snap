// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared - ui - Input should render properly 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex flex-col items-start"
      >
        <label
          class="pb-[5px]"
          for="textarea"
        >
          textarea
        </label>
        <textarea
          class="w-full h-[170px] outline-none border-[0.5px] border-solid border-[#464646] rounded-[7px] px-[13px] py-[8px] resize-none focus:dark:drop-shadow-[0px_1px_17px_#406FCB] dark:text-black-900 disabled:bg-black-200 disabled:text-black-700 disabled:border-black-600 dark:disabled:bg-black-500"
          id="textarea"
          placeholder="This is Textarea"
        />
      </div>
    </div>
  </div>
</body>
`;
